Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        public void InitializeInstrumentSecurityTypeMap()
+        {
+            InstrumentSecurityTypeMap = MapInstrumentToSecurityType(GetInstrumentsAsync());
+        }
+
+        /// <summary>
+        /// Returns true if we're currently connected to the broker
+        /// </summary>
+        public override bool IsConnected
+        {
+            get { return _issuedAt + _lifeSpan > DateTime.Now; }
+        }
+
+        /// <summary>
+        /// Gets all open orders on the account. 
+        /// NOTE: The order objects returned do not have QC order IDs.
+        /// </summary>
+        /// <returns>The open orders returned from Oanda</returns>
+        public override List<Order> GetOpenOrders()
+        {
+            var oandaOrders = GetOrderList();
+
+            var orderList = oandaOrders.Select(ConvertOrder).ToList();
+            foreach (var openOrder in orderList)
+            {
+                _cachedOpenOrdersByOandaOrderId[openOrder.Id] = oandaOrders.FirstOrDefault(oo => oo.id == openOrder.Id);
+            }
+            return orderList;
+        }
+
+        /// <summary>
+        /// Gets all holdings for the account
+        /// </summary>
+        /// <returns>The current holdings from the account</returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            var holdings = GetPositions().Select(ConvertHolding).ToList();
+            return holdings;
+        }
+
+        /// <summary>
+        /// Converts the Oanda position into a QuantConnect holding.
+        /// </summary>
+        /// <param name="position">The position.</param>
+        /// <returns></returns>
+        protected Holding ConvertHolding(Position position)
+        {
+            return new Holding
+            {
+                Symbol = position.instrument,
+                Type = InstrumentSecurityTypeMap[position.instrument],
+                AveragePrice = (decimal)position.avgPrice,
+                ConversionRate = 1.0m,
+                CurrencySymbol = "$",
+                Quantity = position.side == "sell" ? -position.units : position.units
+            };
+        }
+
+
+
+        /// <summary>
+        /// Gets the current cash balance for each currency held in the brokerage account
+        /// </summary>
+        /// <returns>The current cash balance for each currency available for trading</returns>
+        public override List<Cash> GetCashBalance()
+        {
+            var cash = new List<Cash>();
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId;
+            var accountResponse = MakeRequest<Account>(requestString);
+            //TODO figure how exchange rates work in Oanda. probably need to call http://developer.oanda.com/exchange-rates-api/v1/currencies/
+            cash.Add(new Cash(accountResponse.accountCurrency, accountResponse.balance.ToDecimal(), new decimal(1.0)));
+            return cash;
```

Comment:
I think we'll also need to query for any forex holdings, since forex trades are technically currency swaps. we model forex by our CashBook object, but we still provide what IB calls a 'virtual' forex position, so I think we would want to return forex from cash as well as from get account holdings as they're used to set different things. get account holdings could return all the virtual positions, but get cash balance would return the actual balance for each currency, this could be found by aggregating all the positions, but hopefully there's an api for it :)